controllers/machine_controller.go

```diff
@@ -321,6 +321,12 @@ func (r *MachineReconciler) reconcileDelete(ctx context.Context, cluster *cluste
 
 			conditions.MarkTrue(m, clusterv1.DrainingSucceededCondition)
 			r.recorder.Eventf(m, corev1.EventTypeNormal, "SuccessfulDrainNode", "success draining Machine's node %q", m.Status.NodeRef.Name)
+
+			// after draining, wait for volumes to be detached from the node
+			if err := r.waitForVolumeDetach(ctx, cluster, m.Status.NodeRef.Name, m.Name); err != nil {
```

Comment:
Q: Based on your experience, how much this operations could take? 
If this could take some time, what about surfacing a condition so the user is informed that node delation is waiting for this step to happen?

Reply:
@MaxRink can share more, for us it is much shorter than node draining, within seconds.
It happens when we have stateful workloads running on CP itself and during CP rolling upgrade when the kube-controller-manager not functioning for a brief period of time.

Reply:
For basically all CSI providers i worked with its less than a minute after pods are evicted.
Logging it seems like a good idea anyway

Reply:
+1 to surfacing it. I worked with IaaS layers where this would have been stuck forever because VolumeDetachment sometimes didn't work at all, so it's nice to be able to see that.